### PR TITLE
feat(launcher): updates legacy import codebase with new launcher flow

### PR DESCRIPTION
**Pull request by invincibleJai**
updates legacy import codebase flow with new launcher flow

Tracks : 

- https://github.com/openshiftio/openshift.io/issues/3363

### CODE REVIEW

**Comment by fabric8cd** [comment]
@invincibleJai fabric8/fabric8-ui:SNAPSHOT-PR-2839-2 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2839-fabric8-ui.badger.fabric8.io

**Comment by joshuawilson** [comment]
Please coordinate with @aptmac and his PR https://github.com/fabric8-ui/fabric8-ui/pull/2794 that removes the "add codebase" button.

Please make sure this works with the new Dashboards that are Internal right now, that @dlabrecq and @mindreeper2420 made.

**Comment by invincibleJai** [comment]
@dlabrecq pls review this, it tracks https://github.com/openshiftio/openshift.io/issues/3363 based on inputs from Stevan

- It doesn't change any existing flow.
- if new launcher flag is enabled 
1. Codebase card on the home screen should show just "Add Codebase" button if empty not "Import Codebase"
2. Codebase card on the home screen should trigger new launcher on click of "+" and "Add Codebase".
3. In Create - > Codebase -> click of "+ Add Codebase" should trigger new launcher which would replace existing flow.
4. In Create - > Codebase if there are no Codebase,  click of "Add Codebase" should trigger new launcher which would replace existing flow.

Let me know if there is something am not aware of based on https://github.com/fabric8-ui/fabric8-ui/pull/2794

File: src/app/dashboard-widgets/add-codebase-widget/add-codebase-widget.component.html [hunk 1]
```diff
@@ -1,25 +1,40 @@
...
-        <h3>This Space has no Codebases</h3>
-        <div class="f8-blank-slate-main-action">
-          <button id="spacehome-codebases-create-button" class="btn btn-primary btn-lg" (click)="addToSpace.emit()">Add Codebase</button>
+      <f8-feature-toggle featureName="AppLauncher">
+        <div class="f8-blank-slate-card" *ngIf="codebases.length === 0" default-level>
+            <h3>This Space has no Codebases</h3>
+            <div class="f8-blank-slate-main-action">
+              <button id="spacehome-codebases-create-button" class="btn btn-primary btn-lg" (click)="addToSpace.emit()">Add Codebase</button>
+            </div>
+            <div class="f8-blank-slate-main-action">
```
[the remark this answers is not in the document]
Comment (mindreeper2420) [reply src/app/dashboard-widgets/add-codebase-widget/add-codebase-widget.component.html]: The "+" in the upper right triggers the same action as "Add Codebase", while "Import Codebase" triggers the slide-in panel.

File: src/app/dashboard-widgets/add-codebase-widget/add-codebase-widget.component.html [hunk 2]
```diff
@@ -1,25 +1,40 @@
...
         <span id="spacehome-codebases-card-title">Codebases</span>
       </h2>
     </div>
     <div class="card-pf-body f8-card-body">
-      <div class="f8-blank-slate-card" *ngIf="codebases.length === 0">
-        <h3>This Space has no Codebases</h3>
-        <div class="f8-blank-slate-main-action">
-          <button id="spacehome-codebases-create-button" class="btn btn-primary btn-lg" (click)="addToSpace.emit()">Add Codebase</button>
+      <f8-feature-toggle featureName="AppLauncher">
+        <div class="f8-blank-slate-card" *ngIf="codebases.length === 0" default-level>
```
[the remark this answers is not in the document]
Comment (invincibleJai) [reply src/app/dashboard-widgets/add-codebase-widget/add-codebase-widget.component.html]: Ah, am not aware of it. We did changes recently for similar flow where we followed this https://github.com/fabric8-ui/fabric8-ui/pull/2791

**Comment by invincibleJai** [comment]
> From looking at the Feature Flags, it appears that the new Launcher is listed as a Beta feature, whereas the new Dashboards are Experimental features.

@mindreeper2420 as per my understanding both may or may not be in same feature so have handled for both default and user level under `<f8-feature-toggle featureName="Analyze.newSpaceDashboard">`

File: src/app/space/create/codebases/codebases.component.html [hunk 3]
```diff
@@ -1,38 +1,76 @@
...
-            <codebases-item-actions [cheRunning]="cheState && cheState.running" [codebase]="item" [index]="index"></codebases-item-actions>
+          <ng-template #actionTemplate let-item="item" let-index="index" class="codebase-item">
+              <codebases-item-actions [cheRunning]="cheState && cheState.running" [codebase]="item" [index]="index"></codebases-item-actions>
+            </ng-template>
+            <ng-template #expandTemplate let-item="item" let-index="index">
+              <codebases-item-details [codebase]="item"></codebases-item-details>
+            </ng-template>
+          </pfng-list>
+        </f8-feature-toggle>
+        <f8-feature-toggle featureName="AppLauncher">
```
Comment (dlabrecq) [inline src/app/space/create/codebases/codebases.component.html]: I see pfng-list wrapped twice with the AppLauncher feature flag. Can you wrap both pfng-list components with a single feature flag and use user-level and default-level?

File: src/app/dashboard-widgets/add-codebase-widget/add-codebase-widget.component.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (dlabrecq) [reply src/app/dashboard-widgets/add-codebase-widget/add-codebase-widget.component.html]: FYI, I had to use default-level and user-level with the feature flags I recently added.

File: src/app/space/create/codebases/codebases.component.html (hunk 3 above)
Comment (invincibleJai) [reply src/app/space/create/codebases/codebases.component.html]: Sure, Dan.

**Comment by fabric8cd** [comment]
@invincibleJai fabric8/fabric8-ui:SNAPSHOT-PR-2839-3 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2839-fabric8-ui.badger.fabric8.io

File: src/app/dashboard-widgets/add-codebase-widget/add-codebase-widget.component.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (mindreeper2420) [reply src/app/dashboard-widgets/add-codebase-widget/add-codebase-widget.component.html]: I'm still not understanding why both buttons appears, as this is confusing, but as long as production will not see both, them I'm okay with this.

**Comment by invincibleJai** [comment]
@corinnekrych @joshuawilson is it good to be merged?

File: src/app/dashboard-widgets/add-codebase-widget/add-codebase-widget.component.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (corinnekrych) [reply src/app/dashboard-widgets/add-codebase-widget/add-codebase-widget.component.html]: @mindreeper2420 indeed both buttons are confusing but this is what we have atm on prod / prod-preview. with the new design removing the add codebase outlet design all together will make sense and i think this is what this PR is all about.

**Comment by corinnekrych** [comment]
When:
1.user-consent: `released` feature only
2. go to Analyse, select `import codebase` the oulet import is shown but I notice a CSS issue:
screenshot 2018-05-02 09 29 14_preview

@invincibleJai this issue is already on prod-preview tho but if you have any idea how to fix, it would be good!

**Comment by invincibleJai** [comment]
[test]

**Pull request merged**